### PR TITLE
Improve close pr

### PR DESCRIPTION
**Pull request by apelisse**
- Improve text
- Add mentions
- Add expected close date
- Remove stale comments (after PR becomes active again)

### CODE REVIEW

File: mungegithub/mungers/close-stale-pr.go [hunk 1]
```diff
@@ -232,11 +231,47 @@ func closePullRequest(obj *github.MungeObject, inactiveFor time.Duration) {
 	obj.ClosePR()
 }
 
+func getInvolvedUsers(obj *github.MungeObject) []string {
+	var users []string
+
+	var assignee string
+	if obj.Issue.Assignee != nil && obj.Issue.Assignee.Login != nil && *obj.Issue.Assignee.Login != "*" {
```
Comment (bgrant0607) [inline mungegithub/mungers/close-stale-pr.go]: Does the API support multiple assignees yet?

Comment (apelisse) [inline mungegithub/mungers/close-stale-pr.go]: No, I'm currently discussing with @eparis about updating the go-github API for that purpose

Comment (bgrant0607) [inline mungegithub/mungers/close-stale-pr.go]: Looks like it:
https://github.com/google/go-github/blob/master/github/issues.go#L39

BTW, I assume the issue API works for pulls also?

**Comment by bgrant0607** [comment]
LGTM, thanks.

Feel free to merge and extend to support multiple assignees in another PR.

File: mungegithub/mungers/close-stale-pr.go [hunk 2]
```diff
@@ -35,21 +35,20 @@ const (
...
-You can add 'keep-open' label to prevent this from happening again.`
-	warningComment = `This PR hasn't been active in %s. Will be closed in %s.
+%s
 
-You can add 'keep-open' label to prevent this from happening.`
+You can add 'keep-open' label to prevent this from happening again. Or add a comment to keep it open another 90 days`
+	warningComment = `This PR hasn't been active in %s. It will be closed in %s (%s).
+
+%s
+You can add 'keep-open' label to prevent this from happening. Or add a comment to keep it open another 90 days`
```
Comment (rmmh) [inline mungegithub/mungers/close-stale-pr.go]: `. Or` => `, or`

File: mungegithub/mungers/close-stale-pr.go [hunk 3]
```diff
@@ -35,21 +35,20 @@ const (
...
 	remindWarning    = 30 * day
-	closingComment   = `This PR hasn't been active in %s. Feel free to reopen.
+	closingComment   = `This PR hasn't been active in %s. Closing this PR. Please reopen if you would like to work towards merging this change, if/when the PR is ready for the next round of review.
 
-You can add 'keep-open' label to prevent this from happening again.`
-	warningComment = `This PR hasn't been active in %s. Will be closed in %s.
+%s
 
-You can add 'keep-open' label to prevent this from happening.`
+You can add 'keep-open' label to prevent this from happening again. Or add a comment to keep it open another 90 days`
```
Comment (rmmh) [inline mungegithub/mungers/close-stale-pr.go]: `. Or` => `, or`

File: mungegithub/mungers/close-stale-pr.go [hunk 4]
```diff
@@ -35,21 +35,20 @@ const (
...
 )
 
 var (
-	closingCommentRE = regexp.MustCompile(`This PR hasn't been active in \d+ days?\. Feel free to reopen.
-
-You can add 'keep-open' label to prevent this from happening again\.`)
-	warningCommentRE = regexp.MustCompile(`This PR hasn't been active in \d+ days?\. Will be closed in \d+ days?\.
-
-You can add 'keep-open' label to prevent this from happening\.`)
+	closingCommentRE = regexp.MustCompile(`This PR hasn't been active in \d+ days?\..*label to prevent this from happening again.`)
```
Comment (rmmh) [inline mungegithub/mungers/close-stale-pr.go]: `...\d+ days?.*reopen` would match the old comments, too.

**Comment by rmmh** [comment]
There should be a `close-stale-pr_test.go` to exercise the different bits of functionality.

**Comment by apelisse** [comment]
I agree with that. I'll work on it.

File: mungegithub/mungers/close-stale-pr.go [hunk 5]
```diff
@@ -223,20 +221,62 @@ func durationToDays(duration time.Duration) string {
...
+		users = append(users, *assignee.Login)
+	}
+
+	return users
+}
+
+func mentionUsers(users []string) string {
+	var mentions string
+	for _, user := range users {
+		mentions += "@" + user + " "
```
Comment (lavalamp) [inline mungegithub/mungers/close-stale-pr.go]: I would make an array and finish up with strings.Join(blah, " ").

File: mungegithub/mungers/close-stale-pr.go [hunk 6]
```diff
@@ -223,20 +221,62 @@ func durationToDays(duration time.Duration) string {
...
+
+	closeDate := time.Now().Add(closeIn).Format("Jan 2, 2006")
+
 	obj.WriteComment(fmt.Sprintf(
 		warningComment,
 		durationToDays(inactiveFor),
-		durationToDays(closeIn)))
+		durationToDays(closeIn),
+		closeDate,
+		mention))
```
Comment (lavalamp) [inline mungegithub/mungers/close-stale-pr.go]: nit: prefer )) on their own line.
```
  mention,
))
```

**Comment by lavalamp** [comment]
nits are optional/fix in followup. LGTM

**Pull request merged**